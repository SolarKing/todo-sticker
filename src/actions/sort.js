// @flow
import type {SortByType, SortColumn} from '../constant/sort';

export type SortActions =
    | SortByAction
    | SortColumnAction;

export type SortColumnAction = {|
    type: 'sortColumn',
    column: SortColumn
|};

export type SortByAction = {|
    type: 'sortBy',
    sortBy: SortByType
|};

export function settingSortColumn(column: SortColumn): SortColumnAction {
    return {
        type: 'sortColumn',
        column: column
    };
}
