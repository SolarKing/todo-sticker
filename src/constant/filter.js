// @flow
export type DueDateFilter = 'today' | 'all';

export const DueDateFilterEnum = {
    today: 'today',
    all: 'all'
};
