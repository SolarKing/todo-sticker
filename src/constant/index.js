// @flow

const DueDate = {
    today: 'today',
    all: 'all'
};


export const FilterEnum = {
    DueDate
};
