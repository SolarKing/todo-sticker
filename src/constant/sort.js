// @flow
export type SortColumn = 'dueDate' | 'createTime';
export type SortByType = 'desc' | 'asc';

export const SortColumnEnum = {
    dueDate: 'dueDate',
    createTime: 'createTime'
};

export const SortByTypeEnum = {
    desc: 'desc',
    asc: 'asc'
};
