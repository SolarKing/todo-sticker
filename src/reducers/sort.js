// @flow
import type {SortState} from '../states/index';
import {SortByTypeEnum, SortColumnEnum} from '../constant/sort';
import type {SortActions} from '../actions/sort';

const initState: SortState = {
    sortBy: SortByTypeEnum.asc,
    sortColumn: SortColumnEnum.dueDate
};

export default function sort(sortState: SortState = initState, action: SortActions) {
    switch (action.type) {
    case 'sortColumn':
        return {
            ...sortState,
            sortColumn: action.column
        };
    default:
        return sortState;
    }
}
