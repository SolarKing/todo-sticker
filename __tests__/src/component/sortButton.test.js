// @flow
import React from 'react';
import { shallow, type ShallowWrapper } from 'enzyme';
import SortButton from '../../../src/component/sortButton/sortButton';
import { SortColumnEnum, SortByTypeEnum } from '../../../src/constant/sort';
import { Dropdown } from 'antd';

describe('SortButton', () => {
    let wrapper: ShallowWrapper;
    let handleSortByChanged = jest.fn();
    let handleSortColumnSelected = jest.fn();

    beforeAll(() => {
        wrapper = shallow(
            <SortButton
                sortColumn={SortColumnEnum.createTime}
                sortByType={SortByTypeEnum.asc}
                handleSortByChanged={handleSortByChanged}
                handleSortColumnSelected={handleSortColumnSelected}
            />
        );
    });

    it('should render correct', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should handle sort column change', () => {
        let menuClickValue = {
            key : SortColumnEnum.dueDate,
        };
        let menu = shallow(wrapper.find(Dropdown).prop('overlay'));
        menu.simulate('click', menuClickValue);
        expect(handleSortColumnSelected.mock.calls.length).toEqual(1);
        expect(handleSortColumnSelected.mock.calls[0][0]).toEqual(SortColumnEnum.dueDate);
    });

    it('should handle sort rule change', () => {
        wrapper.find('SortRuleSpan').simulate('click');
        expect(handleSortByChanged.mock.calls.length).toEqual(1);
        expect(handleSortByChanged.mock.calls[0][0]).toEqual(SortByTypeEnum.desc);
        
    });

});