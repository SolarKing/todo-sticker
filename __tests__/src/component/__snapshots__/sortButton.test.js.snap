// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortButton should render correct 1`] = `
<div>
  <p
    style={
      Object {
        "display": "inline",
      }
    }
  >
     Sort By 
  </p>
  <Dropdown
    mouseEnterDelay={0.15}
    mouseLeaveDelay={0.1}
    overlay={
      <Menu
        className=""
        onClick={[Function]}
        prefixCls="ant-menu"
        theme="light"
      >
        <MenuItem>
          <a>
            Due Date
          </a>
        </MenuItem>
        <MenuItem>
          <a>
            Create Time
          </a>
        </MenuItem>
      </Menu>
    }
    placement="bottomLeft"
    prefixCls="ant-dropdown"
    trigger={
      Array [
        "click",
      ]
    }
  >
    <a>
      createTime
    </a>
  </Dropdown>
  <SortRuleSpan
    className="glyphicon glyphicon-sort-by-order"
    onClick={[Function]}
  />
</div>
`;
